### PR TITLE
Fix cppcheck 1.87 warnings in qt/ directory

### PR DESCRIPTION
**Pull request by soininen**
This PR fixes most `cppcheck` 1.87 warnings in a selected subdirectories of the `qt/` directory.

**To test:**

Code review.

Fixes partially #22912.

*This does not require release notes* because this is internal code cleanup.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/EnggDiffraction/EnggVanadiumCorrectionsModel.cpp [hunk 1]
```diff
@@ -30,11 +30,7 @@ const std::string EnggVanadiumCorrectionsModel::VANADIUM_INPUT_WORKSPACE_NAME =
     "engggui_vanadium_ws";
 
 std::pair<Mantid::API::ITableWorkspace_sptr, Mantid::API::MatrixWorkspace_sptr>
-EnggVanadiumCorrectionsModel::calculateCorrectionWorkspaces(
-    const std::string &vanadiumRunNumber) const {
-  const auto vanadiumRunName = generateVanadiumRunName(vanadiumRunNumber);
-  const auto vanadiumWS =
```
Comment (Mantid-Matthew) [inline qt/scientific_interfaces/EnggDiffraction/EnggVanadiumCorrectionsModel.cpp]: Despite looking like it is unused I think this puts a workspace in the ADS which is then used later in this function.

File: qt/scientific_interfaces/ISISSANS/SANSRunWindow.cpp [hunk 2]
```diff
@@ -886,8 +886,6 @@ bool SANSRunWindow::loadUserFile() {
     return false;
   }
 
-  const auto settings = getReductionSettings();
```
Comment (Mantid-Matthew) [inline qt/scientific_interfaces/ISISSANS/SANSRunWindow.cpp]: Whilst it looks like the returned value does nothing the getReductionSettings function creates a PropertyManager object and stores it in the PropertyManagerDataService so I think this call may be needed anyway.

File: qt/scientific_interfaces/EnggDiffraction/EnggVanadiumCorrectionsModel.cpp (hunk 1 above)
Comment (soininen) [reply qt/scientific_interfaces/EnggDiffraction/EnggVanadiumCorrectionsModel.cpp]: Indeed it does. Nice catch!

File: qt/scientific_interfaces/ISISSANS/SANSRunWindow.cpp (hunk 2 above)
Comment (soininen) [reply qt/scientific_interfaces/ISISSANS/SANSRunWindow.cpp]: I reverted the change and added a comment so that others will not burn their fingers here.

**Review by Matthew-Andrew — APPROVED** [review]
This looks good to me now. Thanks for taking a look  :+1:

**Comment by martyngigg** [comment]
The changes are qt only so cannot have broken the system test.

**Pull request merged**